textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/source/IndexSearchSourceOperator.java

```diff
@@ -2,27 +2,47 @@
 
 import edu.uci.ics.textdb.api.common.ITuple;
 import edu.uci.ics.textdb.api.dataflow.ISourceOperator;
+import edu.uci.ics.textdb.api.storage.IDataReader;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
 
 /**
  * Created by chenli on 3/28/16.
  */
 public class IndexSearchSourceOperator implements ISourceOperator {
 
+private IDataReader dataReader;
+    
+    public IndexSearchSourceOperator(IDataReader dataReader) throws DataFlowException {

```

Comment:
We discussed this before and we decided that we will have 2 different classes `ScanBasedSourceOperator` and `IndexBasedSourceOperator` but they will use the same helper class internally. The helper class here is `LuceneDataReader`.